Updated docs on how Mautic tracks contacts, fixed cookie expiration, and added list of cookies used by Mautic (#166)
mautic/mautic-documentation: pages/09.contacts/01.manage-contacts/02.contact-monitoring/docs.en.md

Or, copy the code below and change the URL to your Mautic instance.

> Mautic sets cookies with a lifetime of 1 year. Returning visitors are identified exclusively by the cookie. If no cookie exists yet, Mautic creates a new contact and sets the cookie. Make sure your website url is entered in the CORS settings. This is the first step in better contact identification. Note that if a browser is set to not accept cookies, this may result in each hit creating a new visitor. If this behavior is concerning, see Fingerprint option below. 

Reviewer: Couple of small tweaks to make it more readable and applying the styleguide - also removed the reference to fingerprinting which I will remove in a separate PR as it is no longer supported.

```suggestion
Mautic sets cookies with a lifetime of 1 year. Returning visitors are identified exclusively by the cookie. If no cookie exists yet, Mautic creates a new Contact and sets the cookie. 

Make sure your website URL is entered in the CORS settings as explained below. 

Note that if a browser is set to not accept cookies, this may result in each hit creating a new visitor. 
```

});
```

#### Local contact cookie (first party cookie)

> If CORS is configured to allow access from the domain the mtc.js is embedded, a cookie will be placed on the same domain with the name of `mtc_id`. This cookie will have the value of the ID for the currently tracked contact but is not used to track the contact. This provides access to server side software to the contact ID and thus providing the ability to integrate with Mautic's REST API as well.

Reviewer: ```suggestion
If CORS is configured to allow access from the domain where the mtc.js is embedded, a cookie will be placed on the same domain with the name of `mtc_id`. This cookie will have the value of the ID for the currently tracked contact but is not used to track the contact. This enables the server side software to access the contact ID, and thus providing the ability to integrate with Mautic's REST API as well.
```

```
This code will send all the automatic data to Mautic and adds also email and firstname. The values of those fields must be generated by your system. 

> The tracking code also supports company fields. Mautic can assign a company to your tracked contact based on company name. Then you have to add **company** or **companyname** parameter to tracking code along with other companies fields ( companyemail, companyaddress1, companyaddress2, companyphone, companycity, companystate, companyzipcode, companycountry, companywebsite, companynumber_of_employees, companyfax, companyannual_revenue, companyindustry, companyindustry, companydescription...):

Reviewer: ```suggestion
The tracking code also supports Company fields. Mautic can assign a Company to your tracked Contact based on Company name. Then you have to add the **company** or **companyname** parameter to the tracking code, along with other Companies fields (companyemail, companyaddress1, companyaddress2, companyphone, companycity, companystate, companyzipcode, companycountry, companywebsite, companynumber_of_employees, companyfax, companyannual_revenue, companyindustry, companyindustry, companydescription...):
```

Contact tags and UTM codes can also be used. 

```
    mt('send', 'pageview', {email: 'my@example.com', firstname: 'John', company: 'Mautic', companyemail: 'mautic@example.com', companydescription: 'description of company', companywebsite: 'https://example.com', tags: 'addThisTag,-removeThisTag', utm_campaign: 'Some Campaign'});

```

> #### Load event

Reviewer: ```suggestion
#### Load Event
```

#### Load Event
#### Load event

> As the JS tracking request is loaded asynchronously, you can ask JS to call a function when a request is loaded. To do that, define a *onload* function in options like this:

Reviewer: ```suggestion
As the JS tracking request is loaded asynchronously, you can ask JS to call a function when a request is loaded. To do that, define a *onload* function in options like this:
```

Reviewer 2: There doesn't seem to be a change in this?

Reviewer: #### Load Event
#### Load event

It was this - but somehow the comment went wonky! Capitalising the E.

Mautic currently supports `page_url`, `referrer`, `language`, and `page_title` (note that the use of `url` and `title` are deprecated due to conflicts with contact fields).

#### UTM Codes 
 
Currently, `utm_medium`, `utm_source`, `utm_campaign`, `utm_content`, and `utm_term` are used to generate the content in a new timeline entry.

> `utm_campaign` will be used as  the timeline entry's title.

Reviewer: ```suggestion
`utm_campaign` will be used as the timeline entry's title.
```

Please note that UTM tags are recorded only on a form submission that contains the action "Record UTM Tags".

#### Contact Fields

> You can also pass information specific to your contact by setting Mautic contact field(s) to be publicly updatable. Note that values appended to the tracking pixel should be url encoded (%20 for spaces, %40 for @, etc).

Reviewer: ```suggestion
You can also pass information specific to your Contact by setting Mautic Contact field(s) to be publicly updatable. Note that values appended to the tracking pixel should be url encoded (%20 for spaces, %40 for @, etc).
```

#### Tags

> The contact's tags can be changed by using the `tags` query parameter. Multiple tags can be separated by comma. To remove a tag, prefix it with a dash (minus sign).  

Reviewer: ```suggestion
The Contact's Tags can be changed by using the `tags` query parameter. Multiple Tags can be separated by comma. To remove a Tag, prefix it with a dash (minus sign).  
```

> For example, `mtracking.gif?tags=ProductA,-ProductB` would add the ProductA tag to the contact and remove ProductB.

Reviewer: ```suggestion
For example, `mtracking.gif?tags=ProductA,-ProductB` would add the ProductA Tag to the Contact and remove ProductB.
```

### Identify visitors by tracking URL

There is a configuration section for identifying visitors by tracking URL although this is not recommended for use because it could be used to spoof tracking. If enabled, returning visitors will be identified by tracking URLs from channels (especially from emails) when no cookie exists yet.

Note: The email contact field has to be marked as a unique identifier and publicly updatable in your Mautic configuration.

> #### How are contacts tracked with the tracking script?

Reviewer: ```suggestion
#### How are Contacts tracked with the tracking script?
```

> When using the tracking script, contacts are tracked with third party cookies on the Mautic instance's domain and/or the browser's local storage.

Reviewer: ```suggestion
When using the tracking script, Contacts are tracked with third party cookies on the Mautic instance's domain and/or the browser's local storage.
```

Although the script will write first party cookies to the tracked domain (expires with the session), they are NOT used for tracking. See "Local contact cookie (first party cookie)."

> When a contact visits the website for the first time, the tracking script will make a call to Mautic. Mautic will check if `mautic_device_id` cookie is set on it's domain. If it exists and if the device_id is found in Mautic's database, Mautic will recognize the request as the contact associated with the given device.

Reviewer: ```suggestion
When a Contact visits the website for the first time, the tracking script will make a call to Mautic. Mautic will check if the `mautic_device_id` cookie is set on it's domain. If it exists and if the device_id is found in Mautic's database, Mautic will recognize the request as the Contact associated with the given device.
```

> Mautic will return the contact ID, the device ID, and a legacy session ID (this is the same as the device ID). These value are stored in the browser's local storage (if applicable) and written to the site's domain as a first party cookie (not used for tracking).

Reviewer: ```suggestion
Mautic will return the Contact ID, the device ID, and a legacy session ID (this is the same as the device ID). These value are stored in the browser's local storage (if applicable) and written to the site's domain as a first party cookie (not used for tracking).
```

> The next time the tracking script sends a request to Mautic, it will use the device ID from the browser's local storage to identify the tracked contact. If that cannot be found, Mautic will default to the cookies stored on it's own domain (so third party cookies) to identify the contact.

Reviewer: ```suggestion
The next time the tracking script sends a request to Mautic, it will use the device ID from the browser's local storage to identify the tracked Contact. If that cannot be found, Mautic will default to the cookies stored on it's own domain (so third party cookies) to identify the Contact.
```

#### Steps in Mautic

1. Make the email field publicly updateable, this means that a call to the tracking GIF with the variable email will get properly recognized by Mautic.
1. Make the email field publicly updatable, this means that a call to the tracking GIF with the variable email will get properly recognized by Mautic.

> 2. Set up a form, which will be the access point of your campaign (e.g. a new contact email). Make this form as simple as you can, as you will be POST-ing to it from your App. The typical form URL you will POST to is

Reviewer: ```suggestion
 2. Set up a Form, which will be the access point of your Campaign (e.g. a new Contact email). Make this form as simple as you can, as you will be POST-ing to it from your App. The typical Form URL you will POST to is
```

[TYPO3]: <https://github.com/mautic/mautic-typo3>
[Concrete5]: <https://github.com/mautic/mautic-concrete5>
[Grav]: <https://github.com/mautic/mautic-grav>

### Cookies used by Mautic

> This is a list of cookies potentially used by Mautic when tracking contacts. Note that if using the tracking script, the browser's local storage is used to store a device ID used to track the contact.  

Reviewer: ```suggestion
This is a list of cookies potentially used by Mautic when tracking Cotacts. Note that if using the tracking script, the browser's local storage is used to store a device ID used to track the Contact.  
```

#### Third party cookies
|Name|Expiration|Used by Mautic for tracking?|Description
|--- |--- |--- |---

> |mautic_device_id|1 year|Yes|Used by Mautic to track the contact for either the tracking pixel or if the same key is not found in the brower's local storage for the monitored site.

Reviewer: ```suggestion
|mautic_device_id|1 year|Yes|Used by Mautic to track the Contact for either the tracking pixel or if the same key is not found in the brower's local storage for the monitored site.
```

> |mtc_id|session|No|Stores the Mautic ID of the tracked contact. It was used for tracking prior to Mautic 2.13 but is no longer used and kept for BC reads.

Reviewer: ```suggestion
|mtc_id|session|No|Stores the Mautic ID of the tracked Contact. It was used for tracking prior to Mautic 2.13 but is no longer used and kept for BC reads.
```

> |mautic_referer_id|session|Yes|Stores a reference to the last tracked page for the contact and used by Mautic to determine when a contact exists a page they visited.

Reviewer: ```suggestion
|mautic_referer_id|session|Yes|Stores a reference to the last tracked page for the Contact and used by Mautic to determine when a Contact exists a page they visited.
```

|mtc_sid|session|No|Deprecated cookie that is the same as mautic_device_id. It is no longer actively used by Mautic but kept for BC reads.
|mautic_session_id|unknown|No|Deprecated in Mautic 2 (no longer supported) and removed from Mautic 3

#### First party cookies
|Name|Expiration|Used by Mautic for tracking?|Description
|--- |--- |--- |---

> |mautic_device_id|session|No|Can be used by the monitored site but is not used by Mautic to acdtively track the contact.

Reviewer: ```suggestion
|mautic_device_id|session|No|Can be used by the monitored site but is not used by Mautic to actively track the Contact.
```

> |mtc_id|session|No|tores the Mautic ID for the tracked contact. It is not used for tracking. Can be used by the monitored site to leverage Mautic's REST API on the backend to manipulate the contact.

Reviewer: ```suggestion
|mtc_id|session|No|Stores the Mautic ID for the tracked Contact. It is not used for tracking. Can be used by the monitored site to leverage Mautic's REST API on the backend to manipulate the Contact.
```
